Servers/WindowServer/WSCompositor.cpp

```diff
@@ -77,8 +78,16 @@ void WSCompositor::compose()
         if (wm.any_opaque_window_contains_rect(dirty_rect))
             continue;
         m_back_painter->fill_rect(dirty_rect, wm.m_background_color);
-        if (m_wallpaper)
-            m_back_painter->blit(dirty_rect.location(), *m_wallpaper, dirty_rect);
+        if (m_wallpaper) {
+            auto mode = wm.wm_config()->read_entry("Background", "Mode", "simple");
```

Comment:
Oh, I didn't notice this in the previous review!
WSCompositor::compose() is performance critical and should avoid any unnecessary work.
Let's fetch the background mode somewhere else, and cache it in an enum variable.
That way we don't have to do a bunch of lookups and string comparisons for every rendered frame :)

Reply:
Good point! lol